app/models/manageiq/providers/cloud_manager.rb

```diff
@@ -31,8 +31,8 @@ class << model_name
     has_many :host_aggregates,               :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_networks,                :through     => :network_manager
     has_many :security_groups,               :through     => :network_manager
-
-    has_one  :source_tenant, :as => :source, :class_name => 'Tenant'
+    has_one  :source_tenant, :as => :source, :class_name  => 'Tenant'
+    has_many :vm_or_template_labels,         :through     => :vms, :source => :labels
```

Comment:
@bronaghs has to be ``` :through => :vms_and_templates ```, then you will see all the labels of vms and templates


the relation is defined here https://github.com/Ladas/manageiq/blob/73569782d9a997ff529e61b34c04eefb57633a69/app/models/ext_management_system.rb#L37-L37